### PR TITLE
Added error page for handling generic errors and expired sessions

### PR DESCRIPTION
**Pull request by BrendanBeachBCJ**
# Description

Added a error page. This error page is currently only redirected to when an unexpected error happens when trying to validate an org code.

image

image

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Manual testing, snapshot and storybook tests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: SPDBCSC-410 and 411

### CODE REVIEW

File: src/ecrc-frontend/src/components/page/error/Error.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    return (
+      <main>
+        <Header header={header} />
+        <div className="page">
+          <div className="content col-md-8">
+            <h1>Your session has expired.</h1>
+            <br />
+            <p>Please return to the home page and begin your session again.</p>
+            <div className="buttons">
+              <Button button={homeButton} onClick={home} />
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/error/Error.js]: we can just do:
```
onClick={() => setToHome(true)}
```

File: src/ecrc-frontend/src/components/page/error/Error.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+import "../page.css";
+import Header from "../../base/header/Header";
+import Footer from "../../base/footer/Footer";
+import { Button } from "../../base/button/Button";
+
+export default function Error({ page: { header, error } }) {
+  const [toHome, setToHome] = useState(false);
+
+  const home = () => {
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/error/Error.js]: we can remove this function

File: src/ecrc-frontend/src/components/page/error/Error.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  const homeButton = {
+    label: "Home",
+    buttonStyle: "btn ecrc_go_btn",
+    buttonSize: "btn",
+    type: "submit"
+  };
+
+  if (toHome) {
+    return <Redirect to="/criminalrecordcheck/" />;
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/error/Error.js]: should be able to just redirect to `/`

File: src/ecrc-frontend/src/components/page/error/Error.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            <div className="buttons">
+              <Button button={homeButton} onClick={home} />
+            </div>
+          </div>
+        </div>
+        <Footer />
+      </main>
+    );
+  }
+  return (
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/error/Error.js]: not sure we need two full returns, the differences seem simple enough to have conditional rendering within a single return

File: src/ecrc-frontend/src/components/page/error/Error.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      <div className="page">
+        <div className="content col-md-8">
+          <h1>An unknown error has occurred.</h1>
+          <br />
+          <p>
+            The error description is below. If this error persists, please try
+            again later.
+            <br />
+            <br />
+            {JSON.stringify(error)}
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/error/Error.js]: why do we need to stringify this? isn't it being passed as a string (in the props)

File: src/ecrc-frontend/src/components/page/error/Error.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+          </p>
+          <div className="buttons">
+            <Button button={homeButton} onClick={home} />
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/error/Error.js]: same thing here about the `onClick`

File: src/ecrc-frontend/src/components/page/error/Error.stories.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+export const Mobile = () => (
+  <MemoryRouter>
+    <Error page={genericPage} />
+  </MemoryRouter>
+);
+
+export const SessionExpired = () => (
+  <MemoryRouter>
+    <Error page={{ header, error: "session expired" }} />
```
Comment (sdevalapurkar-bcgov) [inline src/ecrc-frontend/src/components/page/error/Error.stories.js]: could use the spread operator here if wanted 😄

File: src/ecrc-frontend/src/components/page/error/Error.js (hunk 5 above)
Comment (BrendanBeachBCJ) [reply src/ecrc-frontend/src/components/page/error/Error.js]: I missed this when I changed error to be a string - I'll remove this.

File: src/ecrc-frontend/src/components/page/error/Error.js (hunk 4 above)
Comment (BrendanBeachBCJ) [reply src/ecrc-frontend/src/components/page/error/Error.js]: Will do

File: src/ecrc-frontend/src/components/page/error/Error.js (hunk 1 above)
Comment (BrendanBeachBCJ) [reply src/ecrc-frontend/src/components/page/error/Error.js]: Sounds good, I'll change this in the places you pointed out

**Pull request merged**